web-components/src/signals/openfoodfacts.ts

```diff
@@ -1,6 +1,77 @@
 import { signal } from "@lit-labs/signals"
+import { SignalMap } from "../utils/signals"
+import { fetchNutrientsOrder } from "../api/openfoodfacts"
+import { NutrientOrderRequest } from "../types/openfoodfacts"
 
-export const BASE_URL = "https://world.openfoodfacts.org/api/v2"
+export type NutrientsOrder = Record<string, { index: number; displayInEditForm: boolean }>
+export const DEFAULT_OPENFOODFACTS_API_URL = "https://world.openfoodfacts.org"
+
+/**
+ * Store the loading state of the taxonomies to avoid multiple requests.
+ */
+const nutrientsOrderPromises = new SignalMap<Promise<any>>({})
 
 // this enable changing open food facts URL globally
-export const openfoodfactsApiUrl = signal(BASE_URL)
+export const openfoodfactsApiUrl = signal(DEFAULT_OPENFOODFACTS_API_URL)
+
+/**
+ * Nutrients order by country code
+ */
+export const nutrientsOrderByCountryCode = new SignalMap<NutrientsOrder>({})
+
+const setOrderOfNutrient = (
+  obj: NutrientsOrder,
+  index: number,
+  nutrient: NutrientOrderRequest
+): number => {
```

Comment:
/**
 * Insert nutrient at index in NutrientsOrder if it's not already there
 *
 * @return the next index
 */